java/src/main/java/ai/rapids/cudf/RmmEventHandler.java

```diff
@@ -22,9 +22,28 @@ public interface RmmEventHandler {
   /**
    * Invoked on a memory allocation failure.
    * @param sizeRequested number of bytes that failed to allocate
+   * @deprecated deprecated in favor of onAllocFailure(long, boolean)
    * @return true if the memory allocation should be retried or false if it should fail
    */
-  boolean onAllocFailure(long sizeRequested);
+  default boolean onAllocFailure(long sizeRequested) {
+    // this should not be called since it was the previous interface,
+    // and it was abstract before.
+    return false;
+  }
+
+  /**
+   * Invoked on a memory allocation failure.
+   * @param sizeRequested number of bytes that failed to allocate
+   * @param isRetry whether this failure happened while retrying an allocation
+   *                that had previously failed
+   * @return true if the memory allocation should be retried or false if it should fail
+   */
+  default boolean onAllocFailure(long sizeRequested, boolean isRetry) {
```

Comment:
I'll do that. Note: I don't think the caller code needs it, but it may be worthwhile to have it.

The calling code really only wants to worry about retries when, in addition to this being a retry, we have depleted the store. So it keeps a counter in my dev branch that increments only when we are doing the last chance device synchronization handling.